### PR TITLE
Add more tracking features

### PR DESCRIPTION
**Pull request by challen-zhou**
1. Enabled more open tracking features such as KCF/TLD/BOOSTING/MEDIAN_FLOW
2. Added 1 more tracking module regression tool.
3. Add tracking dataset support in tracking moddule.
4. Changed tracking strategy to follow image data topic instead of detection topic.

### CODE REVIEW

**Comment by challen-zhou** [comment]
@sharronliu @yechun1 , please help have a review.

**Comment by LewisLiuPub** [comment]
@challen-zhou seems you need to re-format your code via ament check tools. #include orders, indent errors, redundant space lines, etc.

**Comment by yechun1** [comment]
@challen-zhou thanks for your patch. The patch is too big and is a little diffcult for review. Would you please help to split to several commits with different purpose?

**Comment by challen-zhou** [comment]
Thanks @yechun1 and @LewisLiuPub! Let me have a try.

File: object_analytics_node/src/tracker/tracking_manager.cpp [hunk 1]
```diff
@@ -91,23 +94,28 @@ void TrackingManager::detect(const cv::Mat& mat, const object_msgs::msg::Objects
...
+    RCLCPP_INFO(node_->get_logger(), "detected %s [%d %d %d %d] %.0f%%", n.c_str(), droi.x_offset, droi.y_offset,
                  droi.width, droi.height, dobj.probability * 100);
     std::shared_ptr<Tracking> t;
 #pragma omp critical
     {
       /* get matched tracking with the detected object name (class) and its ROI*/
-      t = getTracking(n, tracked_rect);
+      t = getTracking(n, tracked_rect, probability, stamp);
       /* add tracking if new object detected*/
+#if 0
```
Comment (yechun1) [inline object_analytics_node/src/tracker/tracking_manager.cpp]: seems this code will never been implemented when `if 0`, could it be removed?

File: object_analytics_node/src/tracker/tracking_manager.cpp [hunk 2]
```diff
@@ -174,34 +186,59 @@ void TrackingManager::cleanTrackings()
...
+      continue;
+    }
+
+    in_timezone = 1;
+
     /* seek for the one with the same object name (class), and not yet rectified*/
     if (!t->isDetected() && 0 == obj_name.compare(t->getObjName()))
     {
-      cv::Rect2d trect = t->getTrackedRect();
+    //  cv::Rect2d trect = t->getTrackedRect();
```
Comment (yechun1) [inline object_analytics_node/src/tracker/tracking_manager.cpp]: Please remove marked code. or add comments here.

File: object_analytics_node/src/tracker/tracking_manager.cpp [hunk 3]
```diff
@@ -174,34 +186,59 @@ void TrackingManager::cleanTrackings()
...
         tracking = t;
         match = m;
       }
     }
+    
   }
   /* if matching above the threshold, return the tracking*/
-  if (match >= TrackingManager::kMatchThreshold)
+//  if (match >= TrackingManager::kMatchThreshold)
+  if (match > 0.4)
```
Comment (yechun1) [inline object_analytics_node/src/tracker/tracking_manager.cpp]: seems `0.4` is hardcode, does this parameter be defined in kMatchThreshold?

File: object_analytics_launch/launch/analytics_object.py [hunk 4]
```diff
@@ -27,7 +27,6 @@ def launch(launch_descriptor, argv):
     ld.add_process(
         cmd=[
             get_executable_path(package_name=package, executable_name='composition'),
-            '--localization',                     # enable localization feature, optional
```
[the remark this answers is not in the document]
Comment (challen-zhou) [reply object_analytics_launch/launch/analytics_object.py]: Please ignore the change since it's in split stage.

**Comment by challen-zhou** [comment]
The change would be split into pieces, so close this request.

**Pull request closed without merging**